components/basecomponents/SocialMedia.tsx

```diff
@@ -0,0 +1,34 @@
+import Image from 'next/image';
+
+export const SocialMedia = (props: React.HTMLProps<HTMLAnchorElement>): JSX.Element => (
```

Comment:
Why such general props contract? It would sort of make sense if we were passing all of them to the `<a>` element like `...props`, but we are not not and don't need to right now.

If we are only using `className` I think the props type should be limited to that `type Props = { className: string }`

Reply:
@JuroUhlar 

You have right, that we use only className, currently. But better approach is use type from element so we can change this type on `React.HTMLProps<Pick<HTMLAnchorElement, 'className'>>` or use all props in the component. 

`<a href={link} {...props} />...`

Reply:
Fixed: [commit 033502](https://github.com/cesko-digital/movapp/pull/167/commits/0335027f51d3992f7af3dfe4cdcff05e938eb5ee)

Reply:
Why do you think it's better to take the type from HTML attributes? I think that pattern only makes sense when your component is a thin wrapper over a native element: https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase/#wrappingmirroring-a-html-element 
Even it those cases `HTMLProps` is discouraged. 

Our social media component does not render a single HTML element (as the current typing would suggest). It renders an array of icons. Currently, we do not use the className prop at all, so it might be best to just remove it. But if you prefer to leave it there to allow style customization of the links in the future, we should give it a name that reflects that its applied to each link inside the component, something like `linksCustomClass?: string` :)

Reply:
Did you see fix?

I never see in code that if you want use some prop from HTML attributes, that you will create custom type. What does this mean in terms of Typescript? That you create your own type of something that already exists? 👍🏻

`React.ComponentPropsWithoutRef<"button">`
This you can use, if you want use only some specific props.

`React.ButtonHTMLAttributes<HTMLButtonElement>`
This you can use, if you want use standards props.

Prop is here for reasons that we use in more places.
And if you want to discuss about one prop in a component, so I think I'm gonna get tired of it soon.

Reply:
Hmm I guess it's just subjective stylistic preferences, not so important in the grand scheme of things, feel free to merge as is :)